### PR TITLE
Mode command implementation

### PR DESCRIPTION
**Pull request by A0143487X-Kevin**
Change all existing functions to be of 3 types General / Member / Activity Command
Added InvalidCommandModeException

### CODE REVIEW

**Comment by A0143487X-Kevin** [comment]
Halp. too cheem

File: changemode123.txt [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+ import javafx.stage.Stage;
++import seedu.address.commons.core.AppMode;
+ import seedu.address.commons.core.Config;
+ import seedu.address.commons.core.LogsCenter;
+ import seedu.address.commons.core.Version;
+@@ -61,6 +62,8 @@ public class MainApp extends Application {
+ 
+         initLogging(config);
+ 
++        AppMode.init();
```
Comment (ivyyangyq) [inline changemode123.txt]: This shouldn't be in main app.  if in addressbook.java then would be initiated in model.

Comment (ivyyangyq) [reply changemode123.txt]: oh i don't see it in code. nvm.

File: src/main/java/seedu/address/logic/commands/MemberCommand.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import seedu.address.logic.parser.exceptions.InvalidCommandModeException;
+import seedu.address.model.AppMode;
+import seedu.address.model.Model;
+
+/**
+ * Represents a command with hidden internal logic and the ability to be executed.
+ */
+public abstract class MemberCommand extends Command {
+
+    private ArrayList<AppMode.Modes> modeList = new ArrayList <AppMode.Modes> (
```
Comment (ivyyangyq) [inline src/main/java/seedu/address/logic/commands/MemberCommand.java:17]: This is violating abstraction. Command should not know which command mode it should be in AppMode.

**Comment by ivyyangyq** [comment]
I think the AppMode.Modes getCurrMode() method is revealing too much internal details of how AppMode is implemented. Not to mention it makes checking of the modes very difficult in command. 
For me, it would be better if the isValidMode() in Command class is an abstract method with no input and no return. Then it's implemented in subclasses. For ActivityCommand, it would call isModeActivity() that throws InvalidCommandModeException. The message could be kept in ActivityCommand. For MemberCommand, it would just need to call isModeMember(). For general command, it would just be empty.
The exception would be caught in logic manager in execute in the try block using command.isValidMode().

**Pull request merged**